### ISSUE TITLE
method "findAncestor" freezes application

### ISSUE THREAD

**Issue by r3d9u11** [haxeui/haxeui-core#164]
## Expected Behavior
method "findAncestor" freezes application

## Current Behavior
component mus be found by id, css-class and type

## Test app / minimal test case
UI.xml:
```
<?xml version="1.0" encoding="utf-8"?>
<vbox id="vbox1">
	<box id="box1">
		<vbox id="vbox11" styleNames="vbox11_cls">
			<hbox id="hbox11">
				<scrollview id="scollview11">
					<label id="label4" text="Deep label..." />
				</scrollview>
			</hbox>
		</vbox>
	</box>
</vbox>
```
Main.hx
```
import haxe.ui.Toolkit;
import haxe.ui.core.Screen;
import haxe.ui.core.Component;
import haxe.ui.containers.Box;
import haxe.ui.containers.VBox;
import haxe.ui.containers.HBox;
import haxe.ui.components.Label;
import haxe.ui.components.Button;
import haxe.ui.macros.ComponentMacros;

class Main {
	public static function main() {
		Toolkit.init();
		
		var ui:Component = ComponentMacros.buildComponent("Assets/Xml/UI.xml", VBox);
		Screen.instance.addComponent(ui);
		
		var label4 = ui.findComponent("label4", Label);
		
		trace('label4\'s ancestor with id hbox11 is ${label4.findAncestor("hbox11")}');
	}
}
```
[sample.zip](https://github.com/haxeui/haxeui-core/files/1324273/sample.zip)

## Your Environment

* Version used:n haxe 3.4.2, openfl 6.1.2, haxeui-openfl/haxeui-core from git (master), hxcpp 3.4.188, neko 2.1.0
* Operating System and version (desktop or mobile): Linux Mint 18.2 x64
* Link to your project: [full project](https://github.com/r3d9u11/haxe-basics/tree/master/4_ExtLib/4.4_GuiFrameworks/4.4.2_HaxeUI/4.4.2.2_FunctionalCode/4.4.2.2.1_UIAccessFromClass/4.4.2.2.1.1_ComponentBasicMethods)

Tested with hxcpp and neko. Both targets have a similar result: apllication freezed after findAncestor was called.

Thanks.

**Comment by ianharrigan** [comment]
Oh, this feels like a good one. Nice catch!

**Comment by MSGhero** [comment]
This might have been fixed in validation, and #143 takes care of it too.
